azurerm/resource_arm_automation_certificate.go

```diff
@@ -0,0 +1,220 @@
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"base64": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+
+			"is_exportable": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"thumbprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmAutomationCertificateCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Automation.CertificateClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Certificate creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	accName := d.Get("account_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resGroup, accName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Automation Certificate %q (Account %q / Resource Group %q): %s", name, accName, resGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_automation_certificate", *existing.ID)
+		}
+	}
+
+	base64 := d.Get("base64").(string)
+	isExportable := d.Get("is_exportable").(bool)
+	thumbprint := d.Get("thumbprint").(string)
+	description := d.Get("description").(string)
+
+	parameters := automation.CertificateCreateOrUpdateParameters{
+		CertificateCreateOrUpdateProperties: &automation.CertificateCreateOrUpdateProperties{
+			Base64Value:  &base64,
+			IsExportable: &isExportable,
+			Thumbprint:   &thumbprint,
+			Description:  &description,
+		},
+		Name: &name,
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resGroup, accName, name, parameters); err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resGroup, accName, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Automation Certificate '%s' (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmAutomationCertificateRead(d, meta)
+}
+
+func resourceArmAutomationCertificateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Automation.CertificateClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Certificate update.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	accName := d.Get("account_name").(string)
+
+	description := d.Get("description").(string)
+
+	parameters := automation.CertificateUpdateParameters{
+		CertificateUpdateProperties: &automation.CertificateUpdateProperties{
+			Description: &description,
+		},
+		Name: &name,
+	}
```

Comment:
I am going to guess the missing properties here dont' exist in the update struct?

Reply:
Correct, as per: https://docs.microsoft.com/en-us/rest/api/automation/certificate/update

As a side note, the `PUT` (CreateOrUpdate) call does imply support for updating other properties, but I haven't actually validated whether those changes would be accepted by implementation.

Not sure whether this is being overly conservative, but thinking about the typical use cases - the other fields all relate to certificate material, which are all use cases around destroy/replacing the resource anyway.